### DISCUSSION TITLE
Ld_library_path

### DISCUSSION THREAD

**Discussion by boxerab** [community/community#25267] [Actions]
<p>How can I set LD_LIBRARY_PATH in one job so that it persists for all subequent<br>
jobs, for ubuntu VM ?</p>

**Comment by airtower-luna** [comment]
<p>No, there’s no option to set a workflow-wide environment variable dynamically. However I’m confused why you’d want that: Assuming you’re using GitHub hosted runners each job gets a fresh VM. So you’ll have to install whatever you’re installing that needs an <code>LD_LIBRARY_PATH</code> setting in each job. In that case you can just add the environment setup to the installation.</p>

**Comment by boxerab** [comment, marked as answer]
<p>Thanks, <a class="mention" href="/airtower-luna">@airtower-luna</a> . What I meant to say was persisting LD_LIBRARY_PATH across tasks<br>
inside a single job.</p>
<p>This is what I came up with, which seems to work. I do not have to explicitly set LD_LIBRARY_PATH.</p>
<pre><code class="lang-auto">env:
  BUILD_TYPE: Debug
  DATA_ROOT: ${{ github.workspace }}/data
  GENERATOR_PLATFORM:
  LD_LIBRARY_PATH: ${{ github.workspace }}/kdu
```</code></pre>

**Comment by airtower-luna** [comment]
<p>Yes, that should work. I had assumed you needed a dynamically created value, in that case you’d need to write to <code>GITHUB_ENV</code>. :smiley_cat:</p>
